include/fields2cover/swath_generator/swath_generator_base.h

```diff
@@ -14,10 +14,12 @@
 
 namespace f2c::sg {
 
+enum class SwathOverlapType {NO_OVERLAP, END_OVERLAP, MIDDLE_OVERLAP, EVENLY_DISTRIBUTED_OVERLAP};
+
 class SwathGeneratorBase {
  public:
-  bool getAllowOverlap() const;
-  void setAllowOverlap(bool);
```

Comment:
Do not remove functions that people may be using. If they pull the changes, they will also have to change their code, and we don't want to give more work to users than needed.

For the get, I would only return false if it's `NO_OVERLAP`. For the set, if the argument is false, then `NO_OVERLAP`, and if true, I would pick `EVENLY_DISTRIBUTED_OVERLAP`